addons/product_expiry/models/production_lot.py

```diff
@@ -59,30 +60,20 @@ def _onchange_product(self):
     @api.model
     def _alert_date_exceeded(self):
         # if the alert_date is in the past and the lot is linked to an internal quant
-        # log a next activity on the next.production.lot 
-        alert_lot_ids = self.env['stock.production.lot'].search([('alert_date', '<=', fields.Date.today())])
-        mail_activity_type = self.env.ref('product_expiry.mail_activity_type_alert_date_reached').id
-        stock_quants = self.env['stock.quant'].search([
-            ('lot_id', 'in', alert_lot_ids.ids),
-            ('quantity', '>', 0)]).filtered(lambda quant: quant.location_id.usage == 'internal' )
-        lots = stock_quants.mapped('lot_id')
+        # log an activity on the stock.production.lot
+        alert_lot_ids = self.env['stock.production.lot'].search([
```

Comment:
Actually there is a condition about lot being linked to internal quants that you removed (search on stock.quant then retrieve their lot). I am not a stock expert but I suppose we should keep it to avoid changing the functional feature itself.